src/ec/eddsa.rs

```diff
@@ -43,8 +43,14 @@ impl<'a> Ed25519KeyPair {
 
     /// Copies key data from the given slices to create a new key pair.
     /// The arguments are interpreted as little-endian-encoded key bytes.
-    pub fn from_bytes(private_key: &[u8; 32], public_key: &[u8; 32]) -> Ed25519KeyPair {
+    pub fn from_bytes(private_key: &[u8], public_key: &[u8])
+                      -> Result<Ed25519KeyPair, ()> {
         let mut pair = Ed25519KeyPair { private_public: [0; 64] };
```

Comment:
Please put the construction of `pair` underneath the input validation (to minimize its scope and to show more clearly that nothing bad happens with invalid inputs).